csm_web/scheduler/models.py

```diff
@@ -238,20 +202,23 @@ def end_time(self):
                                   second=self.start_time.second)
                 + self.duration).time()
 
+    def day_number(self):
+        return self.DAY_INDEX.index(self.day_of_week)
+
     def __str__(self):
         formatted_time = self.start_time.strftime("%I:%M %p")
         num_minutes = int(self.duration.total_seconds() // 60)
         return f"{self.location} {self.day_of_week} {formatted_time} for {num_minutes} min"
 
     def save(self, *args, **kwargs):
-        self._location = re.sub(self.SPACE_REDUCE_REGEX, ' ', self._location).strip()
+        self.location = re.sub(self.SPACE_REDUCE_REGEX, ' ', self.location).strip()
         super().save(*args, **kwargs)
 
 
 class Override(ValidatingModel):
     # related_name='+' means Django does not create the reverse relation
     spacetime = models.OneToOneField(Spacetime, on_delete=models.CASCADE, related_name="+")
-    overriden_spacetime = models.OneToOneField(Spacetime, on_delete=models.CASCADE)
+    overriden_spacetime = models.OneToOneField(Spacetime, related_name="_override", on_delete=models.CASCADE)
```

Comment:
is there a reason you're keeping the underscore in the name here?

Reply:
never mind, just saw the `hasattr` on `Spacetime`; I think the `is_expired` logic is best handled in the views though

Reply:
I'm not _keeping_ the underscore, I'm _adding_ it. `override` used to be a public field, now we have an `override @property` which returns `None` if there is no override or one exists but it has expired. I did this because it simplifies the serialization of an `Override` when serializing a `Spacetime`.